docs - add gphdfs to pxf migration info

lisakowen opened the pull request:
work-in-progress PR for docs for gphdfs to pxf migration.

created a new "Migrating gphdfs External Tables to PXF" topic that is xref'd from the install guide. this topic:
- lives in the pxf section of the gpdb 6.0 docs
- assumes that the customer performs the migration (set up pxf, create pxf external tables, verify pxf ext tbls, remove gphdfs ext tables, revoke privs) in their 5.x installation

XX in the docs identify areas where more info is needed.

questions:
- do we want to identify a minimum gpdb 5.x (and associated pxf) version for the migration? i.e. one after pxf moved to server-based config?
- what will pxf NOT support that we want to call out? i've identified a few things in the Limitations section.

doc review site link:
- 6.0 pxf section, "migrating gphdfs -> pxf" new topic - http://docs-lisa-gphdfs2pxf-migration.cfapps.io/6-0/pxf/gphdfs-pxf-migrate.html

frankgh commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+4. Identify the Java version installed in your Greenplum Database cluster, and install a supported version of Java if required.
+
+    PXF supports Java versions 8 and 11, `gphdfs` supports Java version 7 (XX and higher?). If your Greenplum Database cluster hosts are running Java 7, upgrade to Java version 8 or 11 as described in [Installing Java for PXF](install_java.html).
```
Do we know if gphdfs supports other Java versions?

frankgh commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+| Description | gphdfs LOCATION Option | pxf LOCATION Option |
+|----------+-------------+-----------------|
+| Parquet schema  | schema | XXX | 
```
Do you need me to look into this?

lisakowen replied:
i just noticed that we haven't documented a parquet option named SCHEMA.  will add that to the docs todo list.

frankgh commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+| Page size | pagesize | PAGE_SIZE | 
+| Row group size | rowgroupsize | ROWGROUP_SIZE | 
+| Parquet version | parquetversion *or* pqversion | PARQUET_VERSION | 
+| Enable a dictionary | dictionaryenable  | XXX | 
```
also here?

lisakowen commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+As you prepare for migrating from `gphdfs` to PXF:
+
+1. Ensure that you are running Greenplum Database version 5.XX or newer. If you are running an older version of Greenplum, you must upgrade to at least version 5.XX.
```
does this mean that we want people to upgrade to greenplum 5.21.[2 or 3] before they migrate to 6.0 if they are using PXF?

lisakowen commented:
i updated the review site, may need to refresh.

not that excited about how i am presenting the configuration mapping info.  will think about that more.

dyozie commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,224 @@
+|----------+---------------+------------|
+| HADOOP_HOME | Environment variable that identifies the Hadoop installation directory | Not applicable; PXF is bundled with the required dependent Hadoop libraries and JARs | 
+| CLASSPATH | Environment variable that identifies the locations of Hadoop JAR and configuration files | Not applicable, PXF automatically includes the Hadoop libraries, JARs, and configuration files that it bundles in the `CLASSPATH`. PXF also automatically includes user-registered dependencies found in the `$PXF_CONF/lib` directory in the `CLASSPATH`. | 
+| gp_hadoop_target_version | Server configuration parameter that identifies the Hadoop distribution | Not applicable, PXF works out-of-the-box with the different Hadoop distros | 
```
this row and next, chane distros -> distributions

dyozie commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,224 @@
+title: Migrating gphdfs External Tables to PXF
+---
+
+You may be using the `gphdfs` external table protocol in a Greenplum Database version 5.x cluster to access data stored in Hadoop. Greenplum Database version 6.0 removes support for the `gphdfs` protocol. To maintain access to Hadoop data in Greenplum 6.0, you must migrate your `gphdfs` external tables to use the Greenplum Platform Extension Framework (PXF). This involves setting up PXF and creat ...
```
Just as a general note, let's try to use Greenplum 5 and Greenplum 6 where possible.  Especially for 6, these instructions would apply to 6.1/6.2, etc.

lisakowen commented:
@frankgh -  this topic was assuming a greenplum 5 source for gphdfs migration.  greenplum 4 to greenplum 6 gphdfs migration is possible, so needed to update this topic.
-  added greenplum 4 migration support (which involves some actions on 4 installation and some actions on 6 installation) to the topic.  
- refactored some of the steps/procedures to more accurately split the tasks for 4.
- added some notes and a new topic for 4to6 gphdfs.  
- other minor/misc edits.

i pushed the changes to the review site, may need to refresh.

lisakowen commented:
it is not clear to me if removing the gphdfs tables and revoking the role privileges is required for all currently supported methods of greenplum 4/5 to 6 data migration (backup/restore?).

frankgh commented:
I looked at it again, and it's looking good. Let me know if I need to review anything else.

lisakowen commented:
i tried out backup/restore on a database with a few gphdfs tables.  does appear to me that removing tables and revoking privileges is required before migrating (restore returns error).

dyozie commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
````diff
@@ -0,0 +1,249 @@
+ALTER ROLE <role_name> NOCREATEEXTTABLE(protocol='gphdfs',type='writable');
+```
+
+## <a id="mig4"></a>Migrating gphdfs External tables from Greenplum 4
````
I'm not sure this section is really necessary given the previous note.  But if you want to keep it I'd migrate it to the previous note.  Having it at the end of the topic seems strange.

lisakowen replied:
i agree having this at the end of the page was not optimal.  i moved it to the greenplum 4 note.

lisakowen commented on gpdb-doc/markdown/pxf/gphdfs-pxf-migrate.html.md.erb:
```diff
@@ -0,0 +1,249 @@
+
+After you determine the equivalent PXF configuration properties, you will:
+
+1. Identify the Java version installed in your Greenplum Database cluster, and install a supported version of Java if required.
```
identifying only java 8 because pxf greenplum 5 doesn't support java 11.
